retrievalmarket/impl/client.go

```diff
@@ -28,11 +28,15 @@ import (
 	"github.com/filecoin-project/go-fil-markets/retrievalmarket/migrations"
 	rmnet "github.com/filecoin-project/go-fil-markets/retrievalmarket/network"
 	"github.com/filecoin-project/go-fil-markets/shared"
-	"github.com/filecoin-project/go-fil-markets/stores"
 )
 
 var log = logging.Logger("retrieval")
 
+type BlockstoreAccessor interface {
+	Get(key string, rootCid cid.Cid) (bstore.Blockstore, error)
```

Comment:
I would make the argument a typed thing, and possibly introduce a type alias to make things super clear. Otherwise there's a chance that the implementation is storing CARs under different keys that they're being requested.

```
type PieceCID = cid.Cid
```

Also don't understand why the key _and the root CID?